it/src/main/java/com/google/cloud/teleport/it/datastream/DefaultDatastreamResourceManager.java

```diff
@@ -0,0 +1,363 @@
+        createdStreamIds.add(streamId);
+
+        return reference;
+    }
+
+    public Stream updateStreamState(String streamId, Stream.State state)
+            throws ExecutionException, InterruptedException {
+        Stream.Builder streamBuilder = Stream.newBuilder()
+                .setName(StreamName.format(projectId, location, streamId))
+                .setState(state);
+
+        FieldMask.Builder fieldMaskBuilder = FieldMask.newBuilder().addPaths(FIELD_STATE);
+
+        UpdateStreamRequest request = UpdateStreamRequest.newBuilder()
+                .setStream(streamBuilder)
+                .setUpdateMask(fieldMaskBuilder)
+                .build();
+
+        return datastreamClient.updateStreamAsync(request).get();
+    }
+
+    @Override
+    public Stream startStream(String streamId) throws ExecutionException, InterruptedException {
+        return updateStreamState(streamId, Stream.State.RUNNING);
+    }
+
+    @Override
+    public Stream pauseStream(String streamID) throws ExecutionException, InterruptedException {
+        return updateStreamState(streamID, Stream.State.PAUSED);
+    }
+
+    @Override
+    public void cleanupAll() {
+        try {
+            for (String stream : createdStreamIds) {
+                datastreamClient.deleteStreamAsync(DeleteStreamRequest
+                        .newBuilder()
+                        .setName(StreamName.format(projectId, location, stream))
+                        .build()).get();
+            }
+
+            for (String connectionProfile : createdConnectionProfileIds) {
+                datastreamClient.deleteConnectionProfileAsync(DeleteConnectionProfileRequest
+                        .newBuilder().setName(ConnectionProfileName.format(projectId, location, connectionProfile))
+                        .build()).get();
+            }
+        } catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        } catch (ExecutionException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+
+    /**
+     * Builder for {@link DefaultDatastreamResourceManager}.
+     */
+    public static final class Builder {
+        private final String projectId;
+        private final String location;
+        private DatastreamClient datastreamClient;
+        private SourceType sourceType;
+        private JDBCSource jdbcSource;
+
+        private Builder(String projectId, String location) {
+            this.projectId = projectId;
+            this.location = location;
+        }
+
+        public Builder datastreamClient(DatastreamClient datastreamClient) {
+            this.datastreamClient = datastreamClient;
+            return this;
+        }
+
+        public Builder setSourceType(SourceType sourceType) {
+            this.sourceType = sourceType;
+            return this;
+        }
+
+        public Builder setJDBCSource(JDBCSource jdbcSource) {
+            this.jdbcSource = jdbcSource;
+            return this;
+        }
```

Comment:
Just leaving a comment to go alongside my other comments about removing these setters.